### PR TITLE
Feature/ch1391/user permission for purchase cost

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description

Fixes # (issue)
support issue 7798
adds a minimum/floor value to depreciation settings. 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

QUESTION: my assumption right now is that this value only has a roll for the linear depreciation value method
Should It be included in the half year depreciation method as well? @uberbrady 

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/Http/Controllers/DepreciationsController.php [hunk 1]
```diff
@@ -69,6 +69,7 @@ public function store(Request $request)
         // Depreciation data
         $depreciation->name             = $request->input('name');
         $depreciation->months           = $request->input('months');
+        $depreciation->min_value        = $request->input('min-value');
```
Comment (snipe) [inline app/Http/Controllers/DepreciationsController.php:72]: Please use underscores not dashes for input field names for consistency

File: app/Http/Controllers/DepreciationsController.php [hunk 2]
```diff
@@ -127,7 +128,7 @@ public function update(Request $request, $depreciationId = null)
         // Depreciation data
         $depreciation->name      = $request->input('name');
         $depreciation->months    = $request->input('months');
-
+        $depreciation->min_value = $request->input('min-value');
```
Comment (snipe) [inline app/Http/Controllers/DepreciationsController.php:131]: Here as well :)

File: resources/assets/less/sidebar.less [hunk 3]
```diff
@@ -19,7 +19,7 @@
     width @transition-speed @transition-fn;
   .transition-transform(@transition-rule);
   @media (max-width: @screen-header-collapse) {
-      padding-top: 100px;
```
Comment (snipe) [inline resources/assets/less/sidebar.less:22]: Should this maybe be in a different PR?

File: resources/views/depreciations/edit.blade.php [hunk 4]
```diff
@@ -22,5 +22,15 @@
     </div>
     {!! $errors->first('months', '<span class="alert-msg" aria-hidden="true"><i class="fa fa-times" aria-hidden="true"></i> :message</span>') !!}
 </div>
+<!-- Min Depreciation -->
+    <label for="min-value" class="col-md-3 control-label">
+        {{ trans('admin/depreciations/general.min-value') }}
+    </label>
+        <div class="col-md-2" style="padding-left:0px">
+            <input class="form-control" type="number" step="0.01" min="0.00" name="min-value" id="min-value" value="{{ Request::old('min_value', $item->min_value) }}" style="width: 80px;" />
```
Comment (snipe) [inline resources/views/depreciations/edit.blade.php:30]: Looks like a conflict here in the `name="min-value" ` vs `old('min_value`)

Comment (snipe) [reply resources/views/depreciations/edit.blade.php:30]: Also I assume this is doing a number-stepper thing? (A screenshot of what this looks like on your local would be helpful) - we probably don't want a number stepper if this is for a currency - it can be a clunky UX to make people click 100 times if they want the number to be 100, etc.

**Review by snipe — CHANGES_REQUESTED** [review]
I think there might be two unrelated PRs in here? Also not sure why package-lock.json would have changed?

@uberbrady - since this touches depreciation, can you take a look at this too?

**Comment by Godmartinz** [comment]
Shoot! those css changes are from a different branch..I will get the variables consistent.

**Comment by Godmartinz** [comment]
This pull request is going to master. Closing this PR.
